### ISSUE TITLE
Cache pgid

### ISSUE THREAD

**Issue by AstraLuma** [xonsh/slug#11]
`Process` (POSIX version) should somehow cache `pgid` so it's still available after the process exits.

**Comment by AstraLuma** [comment]
Unless it should return `None` instead?

**Comment by ghost** [comment]
cache it in the `ProcessGroup`?

**Comment by AstraLuma** [comment]
Yeah. Currently, after the process exits, the pgid is no longer available.
